app/scaffold-controllers/navigationHeaderController.js

```diff
@@ -21,20 +25,36 @@ function(
     };
 
     NavigationHeaderController.init = function(counterBadgeController) {
-        return HeaderBarPlugin.init().then(function(headerBar) {
+        var generateSearchIcon = function() {
+            return ImageViewPlugin.init().then(function(searchIcon) {
+                searchIcon.setImagePath(HeaderConfig.searchHeaderContent.imageUrl);
+                return searchIcon;
+            });
+        };
+
+        var generateCartIcon =
```

Comment:
For me I think the naming makes sense. The `counterBadgeController.generatePlugin` function returns a counterBadgePlugin which contains the cart plugin as it's image. I think it makes most sense to think of this as the cart icon (which includes the badge functionality). Maybe `generateBadgedCartIcon` or `generateCartIconWithBadge` if you think that is more clear?